lib/phoenix/live_dashboard/live/page_live.ex

```diff
@@ -39,38 +53,126 @@ defmodule Phoenix.LiveDashboard.PageLive do
     quote bind_quoted: [opts: opts] do
       import Phoenix.LiveView
       import Phoenix.LiveView.Helpers
-      import Phoenix.LiveDashboard.LiveHelpers
+      import Phoenix.LiveDashboard.Helpers
       @behaviour Phoenix.LiveDashboard.PageLive
+      @default_refresh 5
+      @supported_refresh [{"1s", 1}, {"2s", 2}, {"5s", 5}, {"15s", 15}, {"30s", 30}]
 
       refresher? = Keyword.get(opts, :refresher?, true)
+      refresh = Keyword.get(opts, :refresh, @default_refresh)
+      refresh_options = Keyword.get(opts, :refresh_options, @supported_refresh)
 
       def __page_live__(:refresher?) do
         unquote(refresher?)
       end
+
+      def __page_live__(:refresh) do
+        unquote(refresh)
+      end
+
+      def __page_live__(:refresh_options) do
+        unquote(refresh_options)
+      end
     end
   end
 
   @impl true
   def mount(%{"node" => _, "page" => page} = params, session, socket) do
-    if module = session[page] do
-      refresher? = module.__page_live__(:refresher?)
+    case Map.fetch(session, page) do
+      {:ok, {module, page_session}} ->
+        assign_mount(socket, module, page_session, params, session)
+
+      {:ok, value} ->
+        msg = "invalid value: #{inspect(value)} must be `{ModulePage, session}`"
+        raise Phoenix.LiveDashboard.PageNotFound, msg
```

Comment:
I think we can remove this branch. We will always validate the session before, it most likely will never be triggered. :)

Reply:
Done